### ISSUE TITLE
Scoped search keys and analytics rule

### ISSUE THREAD

**Issue by elhe26** [typesense/typesense#1509]
## Description

Will scoped search keys work with analytics rules? What I mean by this is if Scoped search keys will also affect analytics rules, meaning that rules will aggregate both analytics purposes and query suggestions based on Scoped search keys and be retrieved in the same manner.

## Steps to reproduce

1. Create Scoped search keys;
2.  Create collection for aggregation;
3.  Create an analytics rule.

## Expected Behavior

All analytics queries and query suggestions will be filtered based on scoped search keys.

**Typesense Version**: 0.25.2

**OS**: MacOS

**Comment by elhe26** [comment]
Hey @kishorenc, is this possible?

**Comment by kishorenc** [comment]
The analytics aggregation happens globally and has no concept of filters. This includes the filters inside a scoped api key. Can you please elaborate on your use case?

**Comment by elhe26** [comment]
@kishorenc Of course. I want to filter `popular` queries by users (sales, marketing, etc...). I thought this could be accomplished by using scoped keys.

**Comment by kishorenc** [comment]
Scoped search key will embed a filter clause. But for user segmentation like sales or marketing, that will depend on the user context right? Or would sales people be searching a specific subset of documents in the collection?

**Comment by elhe26** [comment]
The latter. This will also be the case for some customers using the same collection but only having access to their data. Popular queries will also be filtered by user since customer A is querying products Customer B doesn't have access to.

**Comment by elhe26** [comment]
Any updates?

**Comment by kishorenc** [comment]
We don't have a way to embed filter context into analytics suggestions. This is regardless of the use of scoped API key. Basically what you want is a way to ensure that a particular suggestion is associated to a specific bucket of queries (specified by a filter by clause). This is not possible yet. I will leave this issue open as a feature request.
